### ISSUE TITLE
Planned Analysis: Integrated CNV and SV analyses and chromothripsis

### ISSUE THREAD

**Issue by jharenza** [AlexsLemonade/OpenPBTA-analysis#27]
We have generated CNV output from [ControlFreeC](http://boevalab.com/FREEC/) and [CNVKit](https://cnvkit.readthedocs.io/en/stable/), but are seeking individuals to determine consensus focal calls and/or identify additional algorithms we can run to instill high confidence in focal CNV calls from the WGS dataset.

**Comment by cgreene** [comment]
After https://github.com/AlexsLemonade/OpenPBTA-manuscript/pull/15 is approved and merged, can you write up the CNV methods and file a PR into that subsection so that we can link folks to the current version of the processing code?

It may change in the future, but then we'll have an accurate manuscript-ready description of what was done.

**Comment by jharenza** [comment]
This [machine learning publication](https://genome.cshlp.org/content/29/7/1134.abstract?sid=24e125af-ad75-449a-a342-9d5c2b8dcdd8) may help us with CN true positives:

**Comment by jharenza** [comment]
> After [AlexsLemonade/OpenPBTA-manuscript#15](https://github.com/AlexsLemonade/OpenPBTA-manuscript/pull/15) is approved and merged, can you write up the CNV methods and file a PR into that subsection so that we can link folks to the current version of the processing code?
> 
> It may change in the future, but then we'll have an accurate manuscript-ready description of what was done.

Yes - will work on getting this filled in by the harmonization team.

**Comment by gonzolgarcia** [comment]
**Integrated CNV and SV analyses and chromothripsis.**

The proposed analyses broadly addresses the prevalence and functional impact of structural variation across brain tumors. It is important to note that copy number variations are essentially a subset of structural variants and as such,  both CNV and SV calls are highly overlapping and complementary and should be studied together. I am effectively proposing to merge #27 and #28 issues. 

In order to integrate CNV calls and SV calls we focus on breakpoint co-locallization, more details in the manuscript: https://www.biorxiv.org/content/10.1101/572248v3

Chromothripsis is a catastrophic one time event involving multiple breakpoints and rearrangements of localized regions in the genome. As opposed to chromoplexia, which involve gradually acquired structural variations. Chromothripsis can be identified by a pattern of oscillating copy number states and concomitant structural variants that allow walking through the newly formed chromosome. In practical terms, It can be identified as regions of abnormally high number of CNVs and SVs. 
Different available methods; all of which have limitations:  ShatterSeek (https://github.com/parklab/ShatterSeek), Shatterproof (https://metacpan.org/release/SGOVIND/Shatterproof-0.13) & No-Name (https://www.biorxiv.org/content/10.1101/572248v3)(Focused on regions which SV density is 2 * std. dev above the average of each sample)


**The input format for developing downstream analyses are:**

**CNV segmentation data:**
SampleId, 
chromosome, 
Start, 
End, 
num_probes 		(depreciated, from SNP array format), 
Segment_Mean	(log T/N )

**Allele specific CNV (optional; defining regions of LoH and allelic imbalance)**
SampleId, 
chromosome, 
Start, 
End, 
BAF_mean
Call 		(LOH or AI)

**SV calls file content: (already filtered by Somatic Score; no need to be annotated)**
SampleId, 
Chromosome-origin,
Start-origin, 
End-origin
Chromosome-destination,
Start-destination,
End-destination,
sv_type: 		DEL, DUP, TRA and INV (often divided in head-to-head and tail-to-tail)


**Some proposed readouts and output analyses**

Structural variation. 
1)	A measure of chromosomal instability (CIN) burden (density of breakpoints per Mb; similar to 	tumor mutational burden, TMB) and a plot by tumor type representing CIN burden (this could be compared to TMB).
2)	Recurrently altered genes (perhaps integrated in an  Oncoprint with SNV?)
	For the oncoprint categories:
		- Amplification/tandem-duplication
		- Deep deletion/deletion
		- Other Structural variation: Inversion, translocations
			 
3)	Focus on novel findings… If some newly recurrently altered gene arises will analyze in depth

Chromothripsis:
4)	A barplot with the frequency of chromothripsis prevalence by tumor subtype
5)	A few circus plots with examples of chromothripsis
6)	association of chromothripsis with other somatic alterations (i.e. TP53 status)

Survival analyses  
7)	multivariate analyses including clinical variables as well as overall TMB and chromosomal instability burden and chromothripsis.

**Comment by jharenza** [comment]
merged #27 and #28 here per @gonzolgarcia's request

**Comment by gonzolgarcia** [comment]
**Issue with lumpy data**

As I am trying to filter somatic SVs from the table I realized that the evidence columns "Tumor" and "Normal" are switched. 

In addition, there is no somatic score and haven't found much guidelines for somatic filtering of tumor/normal lumpy results. I will be considering this: https://github.com/arq5x/lumpy-sv/issues/268

**Comment by jharenza** [comment]
Thanks, @gonzolgarcia! You are right, the T/N columns are swapped - we will fix this in V5 release coming next week.

**Comment by guru-yang** [comment]
The Yang Lab will perform analysis on chromothripsis.

**Comment by gonzolgarcia** [comment]
> The Yang Lab will perform analysis on chromothripsis.

Note that there are two callers for CNV (cnvkit & controlfreek) and SV (manta & lumpy)
This dataset still require some further processing and filtering

**Comment by jaclyn-taroni** [comment]
> Note that there are two callers for CNV (cnvkit & controlfreek) and SV (manta & lumpy)
This dataset still require some further processing and filtering

@gonzolgarcia are you planning to generate SV consensus calls?

**Comment by gonzolgarcia** [comment]
> > Note that there are two callers for CNV (cnvkit & controlfreek) and SV (manta & lumpy)
> > This dataset still require some further processing and filtering
> 
> @gonzolgarcia are you planning to generate SV consensus calls?

Before getting a consensus, lumpy requires somatic filtering. It would be nice to have this added to next release

**Comment by jharenza** [comment]
> > > Note that there are two callers for CNV (cnvkit & controlfreek) and SV (manta & lumpy)
> > > This dataset still require some further processing and filtering
> > 
> > 
> > @gonzolgarcia are you planning to generate SV consensus calls?
> 
> Before getting a consensus, lumpy requires somatic filtering. It would be nice to have this added to next release

@guru-yang  - do you have any experience with somatic filtering of LUMPY SVs? The comment referred to here suggests the following:

1) Run SVTyper - docker
2) Filter for somatic calls:
a) keep non-reference SVs in the tumor;
b) keep SVs which have no alternate depth (AO==0) in normal;
c) keep SVs with sufficient depth in the normal (RO>~7)

**Comment by guru-yang** [comment]
> 
> 
> > > > Note that there are two callers for CNV (cnvkit & controlfreek) and SV (manta & lumpy)
> > > > This dataset still require some further processing and filtering
> > > 
> > > 
> > > @gonzolgarcia are you planning to generate SV consensus calls?
> > 
> > 
> > Before getting a consensus, lumpy requires somatic filtering. It would be nice to have this added to next release
> 
> @guru-yang - do you have any experience with somatic filtering of LUMPY SVs? The comment referred to here suggests the following:
> 
>     1. Run SVTyper - docker
> 
>     2. Filter for somatic calls:
>        a) keep non-reference SVs in the tumor;
>        b) keep SVs which have no alternate depth (AO==0) in normal;
>        c) keep SVs with sufficient depth in the normal (RO>~7)

We haven't used LUMPY at all. The filtering steps sounds reasonable. Based on my experience, Manta alone might be good enough for SV calling.

**Comment by gonzolgarcia** [comment]
l

> > > > > Note that there are two callers for CNV (cnvkit & controlfreek) and SV (manta & lumpy)
> > > > > This dataset still require some further processing and filtering
> > > > 
> > > > 
> > > > @gonzolgarcia are you planning to generate SV consensus calls?
> > > 
> > > 
> > > Before getting a consensus, lumpy requires somatic filtering. It would be nice to have this added to next release
> > 
> > 
> > @guru-yang - do you have any experience with somatic filtering of LUMPY SVs? The comment referred to here suggests the following:
> > ```
> > 1. Run SVTyper - docker
> > 
> > 2. Filter for somatic calls:
> >    a) keep non-reference SVs in the tumor;
> >    b) keep SVs which have no alternate depth (AO==0) in normal;
> >    c) keep SVs with sufficient depth in the normal (RO>~7)
> > ```
> 
> We haven't used LUMPY at all. The filtering steps sounds reasonable. Based on my experience, Manta alone might be good enough for SV calling.

You're probably right and manta alone + cnvkit should be enough for Shatterseek?

**Comment by guru-yang** [comment]
> 
> 
> 
> You're probably right and manta alone + cnvkit should be enough for Shatterseek?

Should be enough.

**Comment by jharenza** [comment]
Great! @guru-yang and @gonzolgarcia - you can plan to use Manta + CNVkit for Shatterseek and then we can work on a filtered lumpy data file for release in the next few weeks for general recurrent SV analysis.

**Comment by jharenza** [comment]
@guru-yang and @gonzolgarcia as an update, we are going to remove LUMPY from the release. SVTyper processing is very long per sample (>10 hours), and will require some benchmarking for filtering, which we have de-prioritized in favor of benchmarking copy number. You have both said Manta is fine, so we will drop it. We will have a data release with new CN results coming next week #146, so please let us know if you need help with creating PRs!

**Comment by guru-yang** [comment]
@jharenza Thanks for the update. I am wondering how to get sample metadata. We are able to get gender, age at diagnose, tumor type from Kids First data portal. In order to perform survival analysis, age at last follow up would be needed. Do you know how to get that information? Are there any other information available for the patients, or their parents, such as smoking, alcohol consumption of the parents?

**Comment by cgreene** [comment]
@guru-yang : have you examined the metadata available in the files associated with this project? Once you do, could you file a new issue noting anything that's missing that you'd need for your analysis? Thanks!

**Comment by guru-yang** [comment]
@cgreene I am able to find overall survival in pedcbioportal. Thanks.

**Comment by jaclyn-taroni** [comment]
Hi @guru-yang - overall survival, gender, age at diagnosis, and tumor type are all available in the `pbta-histologies.tsv` file that are part of the data files that are obtained by running the `download-data.sh` script. 

We need people to use that file when putting together their analyses because that ensures that different contributors that are working independently are using the same information across their analyses (e.g., the same overall survival values). If there are additional fields you would like to see in the `pbta-histologies.tsv` file, please file a new issue requesting that information. Thank you!

**Comment by jharenza** [comment]
> @jharenza Thanks for the update. I am wondering how to get sample metadata. We are able to get gender, age at diagnose, tumor type from Kids First data portal. In order to perform survival analysis, age at last follow up would be needed. Do you know how to get that information? Are there any other information available for the patients, or their parents, such as smoking, alcohol consumption of the parents?

@guru-yang as @jaclyn-taroni mentioned, the survival is in the provided histologies file in the data download. We do not have age at last followup in the file currently, but it can be added in the release due next week. Can you please file an issue for that? We have no parental information available, but if there are other things you would like to see from patients, you can also ask in an issue and I can check whether we have the info available.

**Comment by guru-yang** [comment]
@jaclyn-taroni  @jharenza  I see. Thanks a lot. What about smoking and alcohol usage for the probands?  I don't expect smokers in pediatric cohort. Just curious.

**Comment by cgreene** [comment]
@guru-yang : please file a _new_ github issue with requests for metadata so that we can keep this issue, currently titled "Planned Analysis: Integrated CNV and SV analyses and chromothripsis" on that topic. Thanks!

**Comment by jharenza** [comment]
Hi @gonzolgarcia and @guru-yang! When do you think you will be able to file a pull request with either of your analyses? Thanks!

**Comment by guru-yang** [comment]
@jharenza We have made some progress. Is there a regular conference call or similar to share results among the group? Or everything is through github?

**Comment by jaclyn-taroni** [comment]
Hi @guru-yang, great to hear! We encourage you to file pull requests adding the code used to generate results as you have them. The analysis does not need to be complete before getting added to the repository. We have a [pull request template](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/.github/PULL_REQUEST_TEMPLATE.md) with a section for summarizing results to facilitate discussion. You can join the Cancer Data Science Slack `#open-pbta` channel ([more information here](https://github.com/AlexsLemonade/OpenPBTA-analysis#join-the-cancer-data-science-slack)) if you have questions about the pull request model that are better answered in real-time.

**Comment by cgreene** [comment]
I will echo @jaclyn-taroni and @jharenza : please file pull requests adding code as you are writing it. It is much harder to integrate a large amount of code after it is entirely written. Thanks!

**Comment by guru-yang** [comment]
@jharenza @jaclyn-taroni @cgreene Will try to do that soon. I am traveling this week. One quick question, we have seen quite some patients with more than one tumors sequenced. When working on variants, is there a particular strategy to handle these tumors? Such as randomly pick one?

**Comment by jashapiro** [comment]
As of the v7 release, we now provide lists of independent specimens (one tumor per individual) that we would like analyses to use. These are randomly selected, as you suggest, but this allows everyone to use consistent sets. See the bottom of the [Data Formats](https://github.com/AlexsLemonade/OpenPBTA-analysis#data-formats) section of the README for descriptions of those files.

**Comment by guru-yang** [comment]
I noticed in some samples the CNV calls from two algorithms are quite different. I wonder what's the plan going forward. It seems to me generating a consensus CNV call is not easy.

**Comment by jaclyn-taroni** [comment]
Hi @guru-yang - have you taken a look at the copy number consensus issue: https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/128?

**Comment by gonzolgarcia** [comment]
Hello everyone, I wanted to apologize for my lack of contribution to this issue, which I proposed initially. Unfortunately the requirements of my new position at Mount Sinai have let me with very little time bandwidth. For the time being I cannot guaranty that I will contributing steadily to this issue. However, I'd be happy to provide support if still needed as I am working on developing new tools for the integrated analysis of CNVs and structural variations. Best regards to everyone.

**Comment by jaclyn-taroni** [comment]
I filed two more focused issues based on what analyses are in progress vs. those that are not currently accounted for: #393 and #394 

Closing this.
